docs/devel/coding-conventions.md

```diff
@@ -30,12 +30,38 @@ Documentation for other releases can be found at
 <!-- END STRIP_FOR_RELEASE -->
 
 <!-- END MUNGE: UNVERSIONED_WARNING -->
-Coding style advice for contributors
+Code conventions
   - Bash
     - https://google-styleguide.googlecode.com/svn/trunk/shell.xml
   - Go
     - https://github.com/golang/go/wiki/CodeReviewComments
-    - https://gist.github.com/lavalamp/4bd23295a9f32706a48f
+    - Command-line flags should use dashes, not underscores
+    - API conventions
+      - [API changes](api_changes.md)
+      - [API conventions](api-conventions.md)
+    - [Kubectl conventions](kubectl-conventions.md)
+
+Testing conventions
+  - All new packages and most new significant functionality must come with unit tests
+  - Table-driven tests are preferred for testing multiple scenarios/inputs; for example, see [TestNamespaceAuthorization](../../test/integration/auth_test.go)
+  - Significant features should come with integration (test/integration) and/or end-to-end (test/e2e) tests
+    - Including new kubectl commands and major features of existing commands
+
+Directory and file conventions
+    - Avoid package sprawl (see [#4851](http://issues.k8s.io/4851) for discussion)
+      - General utilities belong in new package subdirectories of pkg/util
+    - Directory and document filenames should use dashes, not underscores
+    - Go source files and directories use underscores, not dashes
+    - Third-party code
+      - Third-party Go code is managed using Godeps
+      - Other third-party code belongs in /third_party
+      - Third-party code must include licenses
+      - This includes modified third-party code and excerpts, as well
+
+Coding advice
+  - Go
+    - [Go landmines](https://gist.github.com/lavalamp/4bd23295a9f32706a48f)
```

Comment:
Add a link to https://github.com/kubernetes/kubernetes/blob/master/docs/devel/logging.md